pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -99,38 +98,37 @@ func (ctrl *csiSnapshotSideCarController) storeContentUpdate(content interface{}
 }
 
 // createSnapshot starts new asynchronous operation to create snapshot
-func (ctrl *csiSnapshotSideCarController) createSnapshot(content *crdv1.VolumeSnapshotContent) error {
+func (ctrl *csiSnapshotSideCarController) createSnapshot(content *crdv1.VolumeSnapshotContent) (requeue bool, err error) {
 	klog.V(5).Infof("createSnapshot for content [%s]: started", content.Name)
 	contentObj, err := ctrl.createSnapshotWrapper(content)
 	if err != nil {
 		ctrl.updateContentErrorStatusWithEvent(contentObj, v1.EventTypeWarning, "SnapshotCreationFailed", fmt.Sprintf("Failed to create snapshot: %v", err))
 		klog.Errorf("createSnapshot for content [%s]: error occurred in createSnapshotWrapper: %v", content.Name, err)
-		return err
+		return true, err
```

Comment:
I see in other places, it returns "false, err" without checking if err!=nil, here it returns "true, err". 
What is the recommended value for "requeue" when err != nil? Can you add that in the comments of the function?

Reply:
On error, the value of requeue return val. does not really matter. I added a comment about that to all functions. The other places might have `err == nil`, and then we don't really want to requeue. Here we know there is an error.

I admit the code is not great, I am open for suggestions. I actually thought about returning a specific error that would just cause requeue, but would not cause e.g. events to be sent or error reported in content.status. But it would be even more messy, I think.

Reply:
I think it is better to check if err != nil and set requeue value accordingly in other places as well.  It makes the code easier to read.

Reply:
Ack, I added err != nil checks everywhere.